pkg/backend/aws/ecs.go

```diff
@@ -187,17 +195,111 @@ func (ab *Backend) getNetworkConfig() *types.NetworkConfiguration {
 		securityGroupsPt = append(securityGroupsPt, sgValue)
 	}
 
-	awsvpcConfiguration := &types.AwsVpcConfiguration{
-		AssignPublicIp: types.AssignPublicIpDisabled,
+	awsvpcConfiguration := &ecstypes.AwsVpcConfiguration{
+		AssignPublicIp: ecstypes.AssignPublicIpDisabled,
 		SecurityGroups: securityGroupsPt,
 		Subnets:        privateSubnetsPt,
 	}
-	networkConfig := &types.NetworkConfiguration{
+	networkConfig := &ecstypes.NetworkConfiguration{
 		AwsvpcConfiguration: awsvpcConfiguration,
 	}
 	return networkConfig
 }
 
+func (ab *Backend) Logs(ctx context.Context, serviceName string, since string) error {
+	endTime := time.Now()
+
+	duration, err := time.ParseDuration(since)
+	if err != nil {
+		return errors.Wrapf(err, "invalid duration: '%s'", since)
+	}
+	startTime := endTime.Add(-duration)
+
+	logGroup := ""
+	logStreamName := ""
+
+	// Get a list of task ARNs for a given service
+	taskArns, err := ab.GetTasks(ctx, &serviceName)
+	if err != nil {
+		return errors.Wrapf(err, "error retrieving tasks for service '%s'", serviceName)
+	}
+	tasks, err := ab.GetTaskDetails(ctx, taskArns)
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve task: '%s'", taskArns)
+	}
+
+	taskDefinitions, err := ab.GetTaskDefinitions(ctx, taskArns)
+	if err != nil {
+		return errors.Wrapf(err, "error retrieving task definition for task '%v'", taskArns)
+	}
+	taskDefinitionMap := map[string]ecstypes.TaskDefinition{}
+	for _, taskDefinition := range taskDefinitions {
+		taskDefinitionMap[*taskDefinition.TaskDefinitionArn] = taskDefinition
+	}
+	taskMap := map[string]ecstypes.Task{}
+	for _, task := range tasks {
+		taskMap[*task.TaskArn] = task
+	}
+
+	// For now container name is hardcoded (look across all containers)
+	containerName := ""
+	// Look for the first task definition that has a log group for a matching container name
+	for _, taskArn := range taskArns {
+		taskId, err := ab.getTaskId(taskArn)
+		if err != nil {
+			return errors.Wrapf(err, "invalid task ARN: '%s'", taskArn)
+		}
+
+		task := taskMap[taskArn]
+		taskDefinition := taskDefinitionMap[*task.TaskDefinitionArn]
+		logGroup, logStreamName, err = ab.getLogGroupAndStream(taskDefinition, taskId, containerName)
+		if err != nil {
+			log.Debugf("task definition %s does not have a log group", *taskDefinition.TaskDefinitionArn)
+			continue
+		}
```

Comment:
is this always true for this function when it returns an error? 
also, we lose contextual information from the error here and it is hidden

Reply:
I've added some context. Original code using the aws cli would stream logs without specifying the stream name:

```go
logPrefix := s.backend.Conf().LogGroupPrefix()
	logPath := fmt.Sprintf("%s/%s/%s", logPrefix, stackName, service)

	awsProfile := s.backend.Conf().AwsProfile()
	regionName := "us-west-2"
	awsArgs := []string{"aws", "--profile", *awsProfile, "--region", regionName, "logs", "tail", "--since", since, "--follow", logPath}
```
This code grabs the first valid stream.